tests/bootstrap/test_dns_server.py

```diff
@@ -112,6 +112,47 @@ def test_skip_pattern(
     assert constants.LOCALHOST_IP in stdout.decode().splitlines()
 
 
+def test_skip_pattern_with_surrounding_quotes(
+    docker_network,
+    container_factory: ContainerFactory,
+    stream_container_logs,
+    wait_for_localstack_ready,
+    dns_query_from_container,
+):
+    """
+    Add a skip pattern of localhost.localstack.cloud to ensure that we prioritise skips before
+    local name resolution
+    """
+    ls_container = container_factory(
+        configurators=[
+            ContainerConfigurators.debug,
+            ContainerConfigurators.mount_docker_socket,
+            ContainerConfigurators.network(docker_network),
+            ContainerConfigurators.env_vars(
+                {
+                    "DNS_NAME_PATTERNS_TO_RESOLVE_UPSTREAM": r"'.*localhost.localstack.cloud'",
```

Comment:
To be complete, we should probably include a test with spaces and double quote characters to prevent regressions. And perhaps combinations? 

I don't know if this test should be parametrized over the "troublesome" character, or if we can combine them in one pattern for use here. I think a parametrized test makes more sense. It will slow down execution time a bit, but that is not a problem.

Reply:
I parametrized the original test, less duplication, more covered cases :+1: